clients/fides-js/src/fides.ts

```diff
@@ -85,34 +93,138 @@ declare global {
 // eslint-disable-next-line no-underscore-dangle,@typescript-eslint/naming-convention
 let _Fides: Fides;
 
+/**
+ * Determines if and when to call the API to retrieve geolocation
+ */
+const retrieveEffectiveGeolocation = async (
+  options: FidesOptions,
+  geolocation: UserGeolocation | undefined
+): Promise<UserGeolocation | undefined> => {
+  let effectiveGeolocation;
```

Comment:
hmm wait what happens if `constructLocation` does succeed? it seems like we'd return `undefined` which doesn't seem desirable?

Reply:
If `geolocation` is passed in to `Fides.init()`, there's a possibility it's not in the valid format to be ingested by Fides. Here is the best place to do this validation, because if both `geolocation` AND experience is passed in, we can just disregard `geolocation` altogether, and can continue to render the notices.

Reply:
so `constructLocation` is just a validation safeguard